internal/app/admin/http/cache.go

```diff
@@ -190,6 +185,32 @@ type marshallableCache struct {
 	Cache []marshallableResource
 }
 
+func getRelevantKeys(o *orchestrator.Orchestrator, inputKey string, w http.ResponseWriter) ([]string, error) {
+	var relevantKeys []string
+	// If wildcard suffix provided, retrieve all cache keys that match the given prefix.
+	// If no key is provided, retrieve all keys.
+	if hasWildcardSuffix(inputKey) {
+		// Retrieve all keys
+		allKeys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			fmt.Fprintf(w, "error in getting cache keys: %s", err.Error())
```

Comment:
we should be sending a. marshallableobject back. https://github.com/envoyproxy/xds-relay/blob/master/pkg/marshallable/error.go